### PR TITLE
Remove LargerArrayPool

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Prior to .NET 6 the ArrayPool had a pooling size limit of 2MB, but after .NET 6 this was increased to 1 GB https://github.com/dotnet/runtime/pull/55621 so LargerArrayPool is no longer required.

#### What types of changes does your code introduce?

- [x] Refactoring

#### Requires testing

- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs [hunk 1]
```diff
@@ -240,10 +239,10 @@ public TraceMemory GetTrace()
 
     public void Dispose()
     {
-        if (_memory is not null)
+        byte[] memory = Interlocked.Exchange(ref _memory, null);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs:242]: not really a multi-threaded class?

**Review by LukaszRozmej — APPROVED** [review]
Should EvmPolledMemory be a struct?

**Comment by benaadams** [comment]
> Should EvmPolledMemory be a struct?

Sure

**Comment by Scooletz** [comment]
@benaadams removing my work. The world is truly ending 😭 😉

**Pull request merged**